### ISSUE TITLE
lmod package uses `;` instead of `:` as PATH separator

### ISSUE THREAD

**Issue by eschnett** [LLNL/spack#1046]
The lmod package contains these lines:
```
        # Add our lua to PATH
        os.environ['PATH'] = spec['lua'].prefix.bin + ';' + os.environ['PATH']
```

This does not work; it should use a colon `:` instead of a semicolon `;`. (Alternatively, the lua package should do this already.)

**Comment by alalazo** [comment]
I think I fixed this and updated `lmod` in #1026

**Comment by eschnett** [comment]
Yes it has.
